pkg/plugins/resources/yaml/condition.go

```diff
@@ -3,99 +3,154 @@ package yaml
 import (
 	"errors"
 	"fmt"
+	"strings"
 
 	goyaml "github.com/goccy/go-yaml"
 	"github.com/goccy/go-yaml/parser"
 	"github.com/updatecli/updatecli/pkg/core/pipeline/scm"
-	"github.com/updatecli/updatecli/pkg/core/result"
 
 	"github.com/vmware-labs/yaml-jsonpath/pkg/yamlpath"
 	"gopkg.in/yaml.v3"
 )
 
+var (
+	ErrKeyNotFound = errors.New("key not found")
+)
```

Comment:
WDYT about defining an `errors.go` file in this package and define this as an explicit errors to export it while following usual go conventions?